File: Python/temps_execution.py

# Import
import numpy as np
import numpy.linalg as lg
import scipy.sparse as sp
from time import time
from lasso import *

# Set parameters
m = 50
n = 100
p = 0.1
r = 1
maxiter = 1000
epsilone1 = 1e-6
epsilone2 = 1e-4
maxiter_temps = 50

# Resolution de Ax = b

executions_time = np.zeros(maxiter_temps)
execution_iter = np.zeros(maxiter_temps)
execution_perf = np.zeros(maxiter_temps)

for i in range(maxiter_temps):
	parametres = initialisation(m, n, p)
	start_time = time()
	xk, iter, historique_cout, flags = lasso(*parametres, r, maxiter, epsilone1, epsilone2)
	executions_time[i] = time() - start_time
	execution_iter[i] = iter
	execution_perf[i] = historique_cout[iter-1]
    

# Affichage

print("performance moyenne : {0}".format(execution_perf.mean()))
print("temps moyen d'execution : {0}".format(executions_time.mean()))
print("nombre moyen d'iteration : {0}".format(execution_iter.mean()))

File: Matlab/temps_execution.m

m = 50; %nombre d'individus
n = 100; %nombre d'attributs
p = 0.1; %densite de la matrice A
r=1;
maxiter=1000;
delta=1e-6;
delta0=1e-4;

maxiter_temps = 50;
executions_time = zeros(maxiter_temps,1);
execution_iter = zeros(maxiter_temps,1);
execution_perf = zeros(maxiter_temps,1);

for i=1:maxiter_temps
	[A,b,x0,z0,u0,lambda]=init(m,n,p);
	tic
	[x,h,flag,iter]=lasso(A,b,x0,z0,u0,lambda,r,maxiter,delta,delta0);
	executions_time(i) = toc;
	execution_iter(i) = iter;
	execution_perf(i) = h(iter);
end

disp("performance moyenne : ")
mean(execution_perf)
disp("temps moyen d'execution : ")
mean(executions_time)
disp("nombre moyen d'iteration : ")
mean(execution_iter)